src/AstRelation.h

```diff
@@ -183,39 +200,63 @@ class AstRelation : public AstNode {
         return false;
     }
 
-    /** Operator overload, calls print if reference is given */
-    friend std::ostream& operator<<(std::ostream& os, const AstRelation& rel) {
-        rel.print(os);
-        return os;
+    /** Return i-th clause associated with this relation */
+    AstClause* getClause(size_t idx) const {
+        return clauses[idx].get();
     }
 
-    /** Operator overload, prints name if pointer is given */
-    friend std::ostream& operator<<(std::ostream& os, const AstRelation* rel) {
-        os << rel->getName();
-        return os;
+    /** Obtains a list of the associated clauses */
+    std::vector<AstClause*> getClauses() const {
+        return toPtrVector(clauses);
     }
 
-    /** Print string representation of the relation to a given output stream */
-    void print(std::ostream& os) const override {
-        os << ".decl " << this->getName() << "(";
-        if (!attributes.empty()) {
-            os << attributes[0]->getAttributeName() << ":" << attributes[0]->getTypeName();
+    /** Add a clause to the relation */
+    void addClause(std::unique_ptr<AstClause> clause) {
+        assert(clause && "Undefined clause");
```

Comment:
```suggestion
        assert(clause != nullptr && "Undefined clause");
```

Reply:
Note that `clause` is a `unique_ptr` which has an overloaded bool operation. Thus using it as a bool is entirely valid, I just think that the purpose of the assert is clearer with an explicit nullptr comparison.

Reply:
Done.